File: Resources/TOI-Zero_68/A3/025/025.py
n, w, l = map(int, input().split())
a = [[0] * (w + 1) for _ in range(n)]

for i in range(n):
    c = int(input())
    for _ in range(c):
        k = int(input())
        a[i][k] = 1

for i in range(1, w + 1):
    check = 1
    for j in range(n):
        kk = 0
        for k in range(max(1, i - l), min(w + 1, i + l + 1)):
            kk = max(kk, a[j][k])
        check = min(check, kk)
    
    if check:
        print(1)
        exit()

print(0) 

File: TOI-Zero_68/A3/025/025.cpp
#include<bits/stdc++.h>
using namespace std;

#define ll long long

int main(){
	ios::sync_with_stdio(0);
	cin.tie(0);
	int n,w,l;
    cin >> n >> w >> l;
    int a[n][w+1];
    memset(a,0,sizeof(a));
    for(int i=0;i<n;i++){
        int c;
        cin >> c;
        for(int j=0;j<c;j++){
        	int k;
        	cin >> k;
        	a[i][k]=1;
		}
    }
    for(int i=1;i<=w;i++){
        int check=1;
        for (int j=0;j<n;j++){
            int kk=0;
            for(int k=max(1,i-l);k<=min(w,i+l);k++){
            	kk=max(kk,a[j][k]);
			}
            check=min(check,kk);
        }
        if(check){
        	cout << 1;
        	return 0;
		}
    }
    cout << 0;
}
